### PR TITLE
Ability to disable airspeed scaling

### PR DESCRIPTION
**Pull request by roman-dvorak**
Autogyro (and [rotorcrafts](https://en.wikipedia.org/wiki/Rotorcraft) in general) requires the possibility to disable airspeed scaling.. This pull-request adds a parameter ```FW_ARSP_SCALE``` to the possibility to disable airspeed scaling.

This parameter is two-state. Zero (0) disables airspeed scaling, 1 enables airspeed scaling. 

This issue is solved with setting variable ```_airspeed_scaling``` to 1 for disabling airspeed-scaling. 

https://github.com/ThunderFly-aerospace/PX4Firmware/blob/94feac2b5c96a6f5f102b00dbd7952c179dbab90/src/modules/fw_att_control/FixedwingAttitudeControl.cpp#L267-L268

### CODE REVIEW

**Comment by roman-dvorak** [comment]
Ok, thank you @bresch. I wasn't sure how to do the logic - enable/disable. In new commit is is fixed.

File: src/modules/fw_att_control/fw_att_control_params.c [hunk 1]
```diff
@@ -537,6 +537,20 @@ PARAM_DEFINE_FLOAT(FW_FLAPERON_SCL, 0.0f);
...
+ * Enable of airspeed scaling of control surfaces. Disabling of airspeed scaling
+ * is important for rotor craft and vehicles with rotor wing such as autogyro.
+ * Enable when using aerodynamic control surfaces (e.g.: plane), disable when
+ * using rotor wings (e.g.: autogyro)
+ *
+ * @value 1 Airspeed scaling enabled
+ * @value 0 Airspeed scaling disabled
+ * @group FW Attitude Control
+ */
+PARAM_DEFINE_INT32(FW_ARSP_SCALE, 1);
```
Comment (bresch) [inline src/modules/fw_att_control/fw_att_control_params.c]:
```suggestion
 * Enable airspeed scaling
 *
 * This enables a logic that automatically adjusts the output of the rate controller to take
 * into account the real torque produced by an aerodynamic control surface given
 * the current deviation from the trim airspeed (FW_AIRSPD_TRIM).
 
 * Enable when using aerodynamic control surfaces (e.g.: plane)
 * Disable when using rotor wings (e.g.: autogyro)
 *
 * @boolean
 * @group FW Attitude Control
 */
PARAM_DEFINE_INT32(FW_ARSP_SCALE_EN, 1);
```

File: src/modules/fw_att_control/FixedwingAttitudeControl.hpp [hunk 2]
```diff
@@ -149,6 +149,8 @@ class FixedwingAttitudeControl final : public ModuleBase<FixedwingAttitudeContro
 		(ParamFloat<px4::params::FW_AIRSPD_TRIM>) _param_fw_airspd_trim,
 		(ParamInt<px4::params::FW_ARSP_MODE>) _param_fw_arsp_mode,
 
+		(ParamInt<px4::params::FW_ARSP_SCALE>) _param_fw_arsp_scale,
```
Comment (bresch) [inline src/modules/fw_att_control/FixedwingAttitudeControl.hpp]:
```suggestion
		(ParamInt<px4::params::FW_ARSP_SCALE_EN>) _param_fw_arsp_scale_en,
```

File: src/modules/fw_att_control/FixedwingAttitudeControl.cpp [hunk 3]
```diff
@@ -263,7 +263,9 @@ float FixedwingAttitudeControl::get_airspeed_and_update_scaling()
 	 * Forcing the scaling to this value allows reasonable handheld tests.
 	 */
 	const float airspeed_constrained = constrain(airspeed, _param_fw_airspd_min.get(), _param_fw_airspd_max.get());
-	_airspeed_scaling = _param_fw_airspd_trim.get() / airspeed_constrained;
+
+	// Disable airspeed scaling with parameter
+	_airspeed_scaling = (_param_fw_arsp_scale.get()) ? (_param_fw_airspd_trim.get() / airspeed_constrained) : 1.0f;
```
Comment (bresch) [inline src/modules/fw_att_control/FixedwingAttitudeControl.cpp]:
```suggestion
	_airspeed_scaling = (_param_fw_arsp_scale_en.get()) ? (_param_fw_airspd_trim.get() / airspeed_constrained) : 1.0f;
```

File: src/modules/fw_att_control/FixedwingAttitudeControl.cpp [hunk 4]
```diff
@@ -263,7 +263,9 @@ float FixedwingAttitudeControl::get_airspeed_and_update_scaling()
 	 * Forcing the scaling to this value allows reasonable handheld tests.
 	 */
 	const float airspeed_constrained = constrain(airspeed, _param_fw_airspd_min.get(), _param_fw_airspd_max.get());
-	_airspeed_scaling = _param_fw_airspd_trim.get() / airspeed_constrained;
+
+	// Disable airspeed scaling with parameter
```
Comment (bresch) [inline src/modules/fw_att_control/FixedwingAttitudeControl.cpp]:
```suggestion
```

**Comment by roman-dvorak** [comment]
OK, thank you very much!

File: src/modules/fw_att_control/fw_att_control_params.c [hunk 5]
```diff
@@ -537,6 +537,21 @@ PARAM_DEFINE_FLOAT(FW_FLAPERON_SCL, 0.0f);
  */
 PARAM_DEFINE_INT32(FW_ARSP_MODE, 0);
 
+/**
+ * Enable airspeed scaling
+ *
+ * This enables a logic that automatically adjusts the output of the rate controller to take
+ * into account the real torque produced by an aerodynamic control surface given
+ * the current deviation from the trim airspeed (FW_AIRSPD_TRIM).
+
```
Comment (bresch) [inline src/modules/fw_att_control/fw_att_control_params.c]:
```suggestion
 *
```

Comment (bresch) [reply src/modules/fw_att_control/fw_att_control_params.c]: Formatting issue...

**Comment by xdwgood** [comment]
Why is the rotor situation considered in a fixed-wing attitude controller? Is it for vtol?

**Comment by roman-dvorak** [comment]
Standard VTOL uses aerodynamic surfaces for control. For these drones (which uses aerodynamic control surfaces) you need to have scaling turned on.

This feature (ability of disabling airspeed scaling) is needed for autogyro drones that use FW attitude controllers.

**Pull request merged**